modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/ImportAnimalQTL.pm

```diff
@@ -102,12 +103,12 @@ sub fetch_input {
                   source_status     => 'germline',
                   source_name       => 'Animal_QTLdb', #source name in the variation db
                   source_name_short => 'AnimalQTLdb',  #source identifier in the pipeline
-                  threshold => $self->required_param('threshold_qtl'),
+                  threshold => $threshold,
                   );
 
   #create workdir folder
   my $workdir = $pipeline_dir."/".$source_info{source_name_short}."/".$species;
-  make_path($workdir) or die "Failed to create $workdir $!\n";
+  make_path($workdir) or die "Failed to create $workdir $!\n" unless -e $workdir;
```

Comment:
Suggest using -d to check it is a directory.

For ease of reading, suggest using an if instead of an unless.

If make_path fails due to perm issues, the die message is not reported.

There are examples make_path use with errors e.g. AncestralAlleles/Assign.pm

Reply:
Used -d and error check as in AncestralAlleles in next commit.